src/main/kotlin/com/manimdsl/frontend/AbstractSyntaxTree.kt

```diff
@@ -1,7 +1,17 @@
 package com.manimdsl.frontend
 
 open class ASTNode
-data class ProgramNode(val statements: List<StatementNode>) : ASTNode()
+data class ProgramNode(
+        val functions: List<FunctionNode>,
+        val statements: List<StatementNode>
+) : ASTNode()
+
+data class FunctionNode(
```

Comment:
Can you make all ASTNodes that form part of the algorithm inherit from CodeNode otherwise they cannot show up in the visualisation of the code

Reply:
We need to maintain the the line number the function starts at to jump to it

Reply:
I wasn't sure if it is better to make FunctionNode inherit from CodeNode (because CodeNode inherits StatementNode), or just keep the line number within FunctionNode without overriding?

Reply:
Could we keep in the CodeNode? Just makes IR slightly easier